<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="Mocha Tests" time="0.372" tests="23" failures="12">
  <testsuite name="Root Suite" timestamp="2023-10-12T14:07:46" tests="0" time="0.000" failures="0">
  </testsuite>
  <testsuite name="Posts API" timestamp="2023-10-12T14:07:46" tests="0" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.000" failures="0">
  </testsuite>
  <testsuite name="POST /api/posts/create" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.067" failures="0">
    <testcase name="Posts API POST /api/posts/create should create a new post" time="0.061" classname="should create a new post">
    </testcase>
  </testsuite>
  <testsuite name="GET /api/posts/findAll" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.020" failures="0">
    <testcase name="Posts API GET /api/posts/findAll should get all posts" time="0.018" classname="should get all posts">
    </testcase>
  </testsuite>
  <testsuite name="GET /api/posts/:postId" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.021" failures="1">
    <testcase name="Posts API GET /api/posts/:postId should get a single post by postId" time="0.016" classname="should get a single post by postId">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js:68:31
    at Request.callback (node_modules\superagent\lib\node\index.js:857:12)
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\node_modules\superagent\lib\node\index.js:1070:18
    at IncomingMessage.<anonymous> (node_modules\superagent\lib\node\parsers\json.js:21:7)
    at IncomingMessage.emit (node:events:526:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

      + expected - actual

      -404
      +200
      ]]></failure>
    </testcase>
  </testsuite>
  <testsuite name="Add then Delete" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.013" failures="1">
    <testcase name="Posts API Add then Delete Should delete a post with id 1" time="0.011" classname="Should delete a post with id 1">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js:81:29
    at Request.callback (node_modules\superagent\lib\node\index.js:857:12)
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\node_modules\superagent\lib\node\index.js:1070:18
    at IncomingMessage.<anonymous> (node_modules\superagent\lib\node\parsers\json.js:21:7)
    at IncomingMessage.emit (node:events:526:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

      + expected - actual

      -404
      +200
      ]]></failure>
    </testcase>
  </testsuite>
  <testsuite name="User Controller Tests" timestamp="2023-10-12T14:07:46" tests="0" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.000" failures="0">
  </testsuite>
  <testsuite name="POST /add" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.061" failures="1">
    <testcase name="User Controller Tests POST /add it should create a user" time="0.059" classname="it should create a user">
      <failure message="Cannot read properties of undefined (reading &apos;have&apos;)" type="TypeError"><![CDATA[TypeError: Cannot read properties of undefined (reading 'have')
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js:107:22
    at Request.callback (node_modules\superagent\lib\node\index.js:857:12)
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\node_modules\superagent\lib\node\index.js:1070:18
    at IncomingMessage.<anonymous> (node_modules\superagent\lib\node\parsers\json.js:21:7)
    at IncomingMessage.emit (node:events:526:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)]]></failure>
    </testcase>
  </testsuite>
  <testsuite name="/POST login" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.021" failures="1">
    <testcase name="User Controller Tests /POST login it should login a user" time="0.021" classname="it should login a user">
      <failure message="Cannot read properties of undefined (reading &apos;have&apos;)" type="TypeError"><![CDATA[TypeError: Cannot read properties of undefined (reading 'have')
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js:129:22
    at Request.callback (node_modules\superagent\lib\node\index.js:857:12)
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\node_modules\superagent\lib\node\index.js:1070:18
    at IncomingMessage.<anonymous> (node_modules\superagent\lib\node\parsers\json.js:21:7)
    at IncomingMessage.emit (node:events:526:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)]]></failure>
    </testcase>
  </testsuite>
  <testsuite name="Posts API" timestamp="2023-10-12T14:07:46" tests="0" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.000" failures="0">
  </testsuite>
  <testsuite name="POST /api/orders/create" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.012" failures="0">
    <testcase name="Posts API POST /api/orders/create create a new order" time="0.012" classname="create a new order">
    </testcase>
  </testsuite>
  <testsuite name="GET /api/orders/findAll" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.006" failures="0">
    <testcase name="Posts API GET /api/orders/findAll should retrieve all orders" time="0.006" classname="should retrieve all orders">
    </testcase>
  </testsuite>
  <testsuite name="Get /api/orders/findOne" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.009" failures="1">
    <testcase name="Posts API Get /api/orders/findOne should return 1 exact order" time="0.007" classname="should return 1 exact order">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js:177:33
    at Request.callback (node_modules\superagent\lib\node\index.js:857:12)
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\node_modules\superagent\lib\node\index.js:1070:18
    at IncomingMessage.<anonymous> (node_modules\superagent\lib\node\parsers\json.js:21:7)
    at IncomingMessage.emit (node:events:526:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

      + expected - actual

      -404
      +200
      ]]></failure>
    </testcase>
  </testsuite>
  <testsuite name="PUT /api/orders/:orderId" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.010" failures="1">
    <testcase name="Posts API PUT /api/orders/:orderId Should update the order price" time="0.010" classname="Should update the order price">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js:191:31
    at Request.callback (node_modules\superagent\lib\node\index.js:857:12)
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\node_modules\superagent\lib\node\index.js:1070:18
    at IncomingMessage.<anonymous> (node_modules\superagent\lib\node\parsers\json.js:21:7)
    at IncomingMessage.emit (node:events:526:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

      + expected - actual

      -404
      +200
      ]]></failure>
    </testcase>
  </testsuite>
  <testsuite name="OrderItems API" timestamp="2023-10-12T14:07:46" tests="0" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.000" failures="0">
  </testsuite>
  <testsuite name="POST /api/orderItems/create" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.012" failures="1">
    <testcase name="OrderItems API POST /api/orderItems/create create a new order item" time="0.012" classname="create a new order item">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 500" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 500
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js:217:31
    at Request.callback (node_modules\superagent\lib\node\index.js:857:12)
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\node_modules\superagent\lib\node\index.js:1070:18
    at IncomingMessage.<anonymous> (node_modules\superagent\lib\node\parsers\json.js:21:7)
    at IncomingMessage.emit (node:events:526:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

      + expected - actual

      -500
      +200
      ]]></failure>
    </testcase>
  </testsuite>
  <testsuite name="GET /api/orderItems/findAll" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.006" failures="0">
    <testcase name="OrderItems API GET /api/orderItems/findAll should retrieve all order items" time="0.006" classname="should retrieve all order items">
    </testcase>
  </testsuite>
  <testsuite name="GET /api/orderItems/find/orderitemId" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.008" failures="0">
    <testcase name="OrderItems API GET /api/orderItems/find/orderitemId should retrieve all order items with the same orderid" time="0.008" classname="should retrieve all order items with the same orderid">
    </testcase>
  </testsuite>
  <testsuite name="GET /api/orderItems/findOne/:orderItemId" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.007" failures="1">
    <testcase name="OrderItems API GET /api/orderItems/findOne/:orderItemId should return one exact order item" time="0.005" classname="should return one exact order item">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js:258:31
    at Request.callback (node_modules\superagent\lib\node\index.js:857:12)
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\node_modules\superagent\lib\node\index.js:1070:18
    at IncomingMessage.<anonymous> (node_modules\superagent\lib\node\parsers\json.js:21:7)
    at IncomingMessage.emit (node:events:526:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

      + expected - actual

      -404
      +200
      ]]></failure>
    </testcase>
  </testsuite>
  <testsuite name="PUT /api/orderItems/:orderItemId" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.010" failures="1">
    <testcase name="OrderItems API PUT /api/orderItems/:orderItemId should update the order item quantity" time="0.009" classname="should update the order item quantity">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js:276:31
    at Request.callback (node_modules\superagent\lib\node\index.js:857:12)
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\node_modules\superagent\lib\node\index.js:1070:18
    at IncomingMessage.<anonymous> (node_modules\superagent\lib\node\parsers\json.js:21:7)
    at IncomingMessage.emit (node:events:526:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

      + expected - actual

      -404
      +200
      ]]></failure>
    </testcase>
  </testsuite>
  <testsuite name="DELETE /api/orderItems/:orderItemId" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.006" failures="1">
    <testcase name="OrderItems API DELETE /api/orderItems/:orderItemId should delete an order item" time="0.005" classname="should delete an order item">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js:289:31
    at Request.callback (node_modules\superagent\lib\node\index.js:857:12)
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\node_modules\superagent\lib\node\index.js:1070:18
    at IncomingMessage.<anonymous> (node_modules\superagent\lib\node\parsers\json.js:21:7)
    at IncomingMessage.emit (node:events:526:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

      + expected - actual

      -404
      +200
      ]]></failure>
    </testcase>
  </testsuite>
  <testsuite name="Add then Delete the order ID = 1" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.009" failures="1">
    <testcase name="OrderItems API Add then Delete the order ID = 1 Should delete an order" time="0.008" classname="Should delete an order">
      <failure message="expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404" type="AssertionError"><![CDATA[AssertionError: expected { Object (_events, _eventsCount, ...) } to have status code 200 but got 404
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js:300:35
    at Request.callback (node_modules\superagent\lib\node\index.js:857:12)
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\node_modules\superagent\lib\node\index.js:1070:18
    at IncomingMessage.<anonymous> (node_modules\superagent\lib\node\parsers\json.js:21:7)
    at IncomingMessage.emit (node:events:526:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)

      + expected - actual

      -404
      +200
      ]]></failure>
    </testcase>
  </testsuite>
  <testsuite name="POST /api/posts/create" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.010" failures="0">
    <testcase name="POST /api/posts/create should create a new post" time="0.010" classname="should create a new post">
    </testcase>
  </testsuite>
  <testsuite name="GET /api/products/findAll" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.009" failures="0">
    <testcase name="GET /api/products/findAll should get all products from the database" time="0.009" classname="should get all products from the database">
    </testcase>
  </testsuite>
  <testsuite name="Favorite Posts API" timestamp="2023-10-12T14:07:46" tests="0" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.000" failures="0">
  </testsuite>
  <testsuite name="POST /api/favoritePosts/addPost" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.009" failures="0">
    <testcase name="Favorite Posts API POST /api/favoritePosts/addPost should add a post to favorites" time="0.008" classname="should add a post to favorites">
    </testcase>
  </testsuite>
  <testsuite name="Favorite Posts API" timestamp="2023-10-12T14:07:46" tests="0" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.000" failures="0">
  </testsuite>
  <testsuite name="GET /api/favoritePosts/findAllFavorites" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.014" failures="0">
    <testcase name="Favorite Posts API GET /api/favoritePosts/findAllFavorites should retrieve all favorite posts" time="0.014" classname="should retrieve all favorite posts">
    </testcase>
  </testsuite>
  <testsuite name="Create comment API" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.009" failures="0">
    <testcase name="Create comment API should create a new comment" time="0.009" classname="should create a new comment">
    </testcase>
  </testsuite>
  <testsuite name="Get comment by Post ID" timestamp="2023-10-12T14:07:46" tests="1" file="C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js" time="0.007" failures="1">
    <testcase name="Get comment by Post ID should source comments by post ID" time="0.006" classname="should source comments by post ID">
      <failure message="Target cannot be null or undefined." type="AssertionError"><![CDATA[AssertionError: Target cannot be null or undefined.
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\test.js:443:37
    at Request.callback (node_modules\superagent\lib\node\index.js:857:12)
    at C:\Users\Jack\Documents\Advanced Software Development 3D Project\fantastic-octo-succotash\node_modules\superagent\lib\node\index.js:1070:18
    at IncomingMessage.<anonymous> (node_modules\superagent\lib\node\parsers\json.js:21:7)
    at IncomingMessage.emit (node:events:526:35)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21)]]></failure>
    </testcase>
  </testsuite>
</testsuites>